packages/flutter_tools/test/general.shard/base/os_utils_test.dart

```diff
@@ -3,42 +3,46 @@
 // found in the LICENSE file.
 
 import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
 import 'package:flutter_tools/src/base/os.dart';
-import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/base/signals.dart';
+import 'package:process/process.dart';
 
 import '../../src/common.dart';
-import '../../src/context.dart';
 
 void main() {
   group('OperatingSystemUtils', () {
     Directory tempDir;
+    FileSystem fileSystem;
 
     setUp(() {
-      tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_tools_os_utils_test.');
+      fileSystem = LocalFileSystem.test(signals: Signals.test());
+      tempDir = fileSystem.systemTempDirectory.createTempSync('flutter_tools_os_utils_test.');
     });
 
     tearDown(() {
       tryToDelete(tempDir);
     });
 
-    testUsingContext('makeExecutable', () async {
-      final File file = globals.fs.file(globals.fs.path.join(tempDir.path, 'foo.script'));
+    testWithoutContext('makeExecutable', () async {
+      const Platform platform = LocalPlatform();
+      final OperatingSystemUtils operatingSystemUtils = OperatingSystemUtils(
+        fileSystem: fileSystem,
+        logger: BufferLogger.test(),
+        platform: platform,
+        processManager: const LocalProcessManager(),
+      );
+      final File file = fileSystem.file(fileSystem.path.join(tempDir.path, 'foo.script'));
       file.writeAsStringSync('hello world');
-      globals.os.makeExecutable(file);
+      operatingSystemUtils.makeExecutable(file);
 
       // Skip this test on windows.
-      if (!globals.platform.isWindows) {
+      if (!platform.isWindows) {
```

Comment:
I know this is copying existing functionality, but should we skip earlier if we're on windows?

Reply:
I think this is sort of - badly testing that chmod doesn't throw.

Reply:
Oh, I missed that this was using the real file system. `chmod` is a no-op on Windows in any case...`  void chmod(FileSystemEntity entity, String mode) {}`

Reply:
makeExecutable too:

`  void makeExecutable(File file) {}`